core/src/main/java/com/linecorp/armeria/client/retry/RetryingRpcClient.java

```diff
@@ -112,6 +115,14 @@ private void retry0(int currentAttemptNo, Backoff backoff, ClientRequestContext
                 eventLoop.submit(() -> retry(currentAttemptNo + 1, backoff, ctx, req, action,
                                              responseFuture));
             } else {
+                long responseTimeoutMillis;
+                try {
+                    responseTimeoutMillis = responseTimeoutMillis(ctx);
+                } catch (ResponseTimeoutException e) {
+                    responseFuture.completeExceptionally(e);
+                    return;
+                }
+                nextDelay = responseTimeoutMillis < 0 ? nextDelay : Math.min(nextDelay, responseTimeoutMillis);
```

Comment:
This block of code that calculates `nextDelay` are duplicated in `RetryingHttpClient` and `RetryingRpcClient`. How about:

- making `responseTimeoutMillis(ctx)` a private method and
- introducing a new protected method that is extracted from this code block?